CMake/utility.cmake

```diff
@@ -386,6 +386,51 @@ function(activate_precompiled_headers _PRECOMPILED_HEADER _SOURCE_FILES)
   endif()
 endfunction()
 
+####################################################################################
+# Adds warnings compiler options to the target depending on the category
+# _TARGET Target name
+# _CATEGORY "library" or "unit_tests" or "benchmarks"
+####################################################################################
+macro( add_warning_levels _TARGET _CATEGORY)
```

Comment:
remove function, see my `set_compiler_warnings`

Reply:
I ve seen, it: Can I had the "categories" too?, Such that I can switch for benchmarks?
I agree with you, that this function is only to add some warnings, cmake should